### ISSUE TITLE
selectChoose not working on integration test

### ISSUE THREAD

**Issue by victor95pc** [cibernox/ember-power-select#867]
I made the import on ember power select test helpers, my app name is "juridico-sengerj'

```javascript
import { selectChoose } from 'ember-power-select/test-support/helpers/ember-power-select'
```

But when I ran the test it didn't return nothing only shows "0 assertions of 0 passed, 0 failed.", no log message, anything.

package.json:
```javascript
{
    "ember-ajax": "2.5.6",
    "ember-autoresize": "0.5.16",
    "ember-bootstrap": "0.9.1",
    "ember-bootstrap-cp-validations": "github:victor95pc/ember-bootstrap-cp-validations",
    "ember-browserify": "1.1.10",
    "ember-cli": "2.6.3",
    "ember-cli-app-version": "^1.0.0",
    "ember-cli-babel": "^5.1.6",
    "ember-cli-bootstrap-datepicker": "0.5.5",
    "ember-cli-dependency-checker": "^1.2.0",
    "ember-cli-deploy": "0.5.1",
    "ember-cli-htmlbars": "^1.0.3",
    "ember-cli-htmlbars-inline-precompile": "^0.3.1",
    "ember-cli-inject-live-reload": "^1.4.0",
    "ember-cli-jshint": "^1.0.0",
    "ember-cli-loading-slider": "^1.3.0",
    "ember-cli-mirage": "^0.2.8",
    "ember-cli-qunit": "^1.4.0",
    "ember-cli-release": "^0.2.9",
    "ember-cli-sass": "5.3.1",
    "ember-cli-sentry": "2.4.2",
    "ember-cli-sri": "^2.1.0",
    "ember-cli-uglify": "^1.2.0",
    "ember-component-inbound-actions": "0.0.4",
    "ember-composable-helpers": "1.1.2",
    "ember-cp-validations": "3.1.4",
    "ember-data": "^2.6.0",
    "ember-export-application-global": "^1.0.5",
    "ember-i18n": "4.3.2",
    "ember-i18n-cp-validations": "2.7.2",
    "ember-inline-edit": "0.1.7",
    "ember-load": "0.0.9",
    "ember-load-initializers": "^0.5.1",
    "ember-pagefront": "0.11.1",
    "ember-power-select": "1.0.0-beta.4",
    "ember-query-params-reset": "0.0.2",
    "ember-rapid-forms": "1.0.0-beta7",
    "ember-resolver": "^2.0.3",
    "ember-rl-dropdown": "0.7.0",
    "ember-route-action-helper": "0.3.3",
    "ember-simple-auth": "1.1.0",
    "ember-simple-auth-token": "1.1.1",
    "ember-sinon": "0.6.0",
    "ember-sweetalert": "git+https://github.com/victor95pc/ember-sweetalert.git",
    "ember-text-mask": "4.2.1",
    "ember-truth-helpers": "1.2.0",
    "ember-uploader": "1.1.0",
    "ember-wormhole": "0.4.1",
    "ember-x-editable": "0.3.1",
    "jspdf": "^1.3.2",
    "jspdf-autotable": "^2.3.1",
    "liquid-fire": "^0.24.0",
    "loader.js": "^4.0.1",
    "phantomjs": "^1.9.20",
    "string-just": "git://github.com/victor95pc/js-string-just.git",
    "text-mask-addons": "^3.5.0"
  }

```

**Comment by snewcomer** [comment]
Did you import it in your start-app file?

http://www.ember-power-select.com/docs/test-helpers

**Comment by cibernox** [comment]
`selectChoose` is a test for acceptance test, not for integration. It is expected that it doesn't work.

The addon also provides a few helpers for integration, but for the most part I'd recommend you try https://github.com/cibernox/ember-native-dom-helpers and just use `click` & `fillIn` as you would with any other element.

**Comment by victor95pc** [comment]
I follow this addon, but it's not working too, I setup a test like this

```javascript
import { click, fillIn, find, findAll, keyEvent, triggerEvent } from 'ember-native-dom-helpers';

moduleForComponent('profissional-degrees', 'Integration | Component | profissional-degrees', {
  integration: true
});


test('I can interact with my component', async function(assert) {
  this.render(hbs```
    {{profissional-degrees}}
  ```);

  await fillIn('.some-input');
  await click('.main-button');
  await keyEvent('.other-input', 'keyup', 40); // down arrow
  await triggerEvent('.some-drop-area', 'mouseenter');

  assert.ok(find('.result-of-event-happened'));
  assert.equal(findAll('.result-list-item').length, 3);
})
```

I just copy/paste a example but it result the same problem above, nothing run.

**Comment by snewcomer** [comment]
@victor95pc an ember twiddle would help if you are still stuck after reading through some documentation below :)  https://ember-twiddle.com/

Most likely this is test setup problem (i.e. no `-test.js` at the end of the file name) or perhaps something else.

http://www.ember-power-select.com/docs/test-helpers

**Comment by victor95pc** [comment]
it's very strange, the test run normal, but when I add, it doesn't run, the file is prefixed with "-test" in the end.
```javascript
import { typeInSearch, clickTrigger } from '../../../helpers/ember-power-select';
```

my start-app looks like this
```javascript
import Ember from 'ember';
import Application from '../../app';
import config from '../../config/environment';
import './fill-in-autocomplete';
import './fill-in-summernote-textarea';
import registerPowerSelectHelpers from '../../tests/helpers/ember-power-select';

registerPowerSelectHelpers();

export default function startApp(attrs) {
  let application;

  let attributes = Ember.merge({}, config.APP);
  attributes = Ember.merge(attributes, attrs); // use defaults, but you can override;

  Ember.run(() => {
    application = Application.create(attributes);
    application.setupForTesting();
    application.injectTestHelpers();
  });

  return application;
}
```
Can't find any reason it's not running, the worst part is not even know where it fails.

**Comment by simonihmig** [comment]
Just stumbled upon this. For a project I made a `selectChoose`-ish helper that mimics the semantics of `selectChoose`, but works for integration tests. I find this quite useful for components tests, as it abstracts away the power-select specific internals like identifying the trigger button, clicking that before looking for the the option list etc. 

I just had to update it because of the recent a11y related changes. Would prefer to integrate it to the project, so it can be in-sync with any future changes. @cibernox any reason you did not want to provide one out of the box? If not, I could create a PR...

**Comment by cibernox** [comment]
When I first created the helpers I though that if people was testing EPS in integration test they would want fine-grained control over what is going on, so I reserved the helpers that do everything for acceptance.

If we introduce this helper I'd like to introduce it following the patterns in ember-native-dom-helpers. That is, there is a single helper that works in both integration and acceptance that you have to manually import instead of register as a global.

Basically use Ember Power * like the first-line troops of the new way of testing in ember.

**Comment by victor95pc** [comment]
I really like the new way, it's less magic and more easy to understand, @simonihmig can you send my this repo?

**Comment by simonihmig** [comment]
@cibernox for a component that maybe extends eps fine-grained control makes sense. But I have a number of components where this is not needed, like components that implement a certain form, where I want to test e.g. that an action is called when submitting valid data. So I need a `selectChoose` helper there. (Actually using `ember-cli-page-object`, so I can abstract the type of select (native or eps) away in tests...)

@victor95pc It is in a private repo for a support addon we use internally. But that's the current helper implementation: https://gist.github.com/simonihmig/3d074c1f0661eb9a209d9e8cacb8f4c1
Pretty much copy pasta from original helper, but just using the imported native helpers instead of Ember's acceptance tests helpers. Worked for us so far, but might need some polishing...

**Comment by cibernox** [comment]
@simonihmig Yes, I agree that that kind of helper might come handy. 

I just think that the best approach is having a single helper that works in both integration and acceptance. Besides, that kind of helpers would be something you just import, so you wouldn't have to register it, which is something I've always found annoying.

**Comment by simonihmig** [comment]
Sure, I like that. Anything special I should be aware of to make it also usable in acceptance tests? Should I generally wrap all async stuff in a `run()`?

Maybe I can try to put a PR together, so we can discuss this better on some concrete code?

**Comment by cibernox** [comment]
Wrapping everything in an Ember run is one step. Another is to stop using `andThen` and just use promises (async/await helps a lot on that).

Are you going to emberconf? We could discuss there. I fly tomorrow so I won't have time.

**Comment by snewcomer** [comment]
@victor95pc `import { selectChoose } from 'juridico-sengerj/tests/helpers/ember-power-select'` if modulePrefix in your environment file is `juridico-sengerj`

**Comment by victor95pc** [comment]
Thx everybody I use @simonihmig's gits, if you guys want I can translate this gist to follow the patterns in ember-native-dom-helpers
